tex/generic/pgf/modules/pgfmoduleparser.code.tex

```diff
@@ -11,101 +11,670 @@
+      \pgfparserlet@gobble@space\pgfparserlet@f
+    \fi
+    \pgfparserlet@h
+  }
+\pgfutil@protected\def\pgfparserlet@h#1%
+  {%
+    \ifx\pgfparserlet@argb\bgroup
+      \pgfparser@fi@BTb
+    \fi
+    \pgfutil@secondoftwo
+    {%
+      \def\pgfparserlet@argb{#1}%
+      \ifx\pgfparser@blankspace\pgfparserlet@argb
+        \def\pgfparserlet@argb{blank space \space}%
+      \fi
+    }
+    {\edef\pgfparserlet@argb{\meaning\pgfparserlet@argb}}%
+    \pgfparserlet@i
+      \pgfparserlet@parser@a
+      \pgfparserlet@state@a
+      \pgfparserlet@arga
+      \pgfparserlet@parser@b
+      \pgfparserlet@state@b
+      \pgfparserlet@argb
+  }
+\def\pgfparserlet@i#1#2#3#4#5#6%
+  {%
+    \pgfutil@IfUndefined{\pgfparser@name{#4} #5 #6 type}
+      {%
+        \pgferror
+          {%
+            There is no rule defined for parser '#4' in state '#5' for '#6'.
+            Aborting \string\pgfparserlet
+          }%
+      }
+      {%
+        \pgfparser@initiate@parser{#1}%
+        \pgfparserlet@let{#1}{#2}{#3}{#4}{#5}{#6}{type}%
+        \pgfparserlet@let{#1}{#2}{#3}{#4}{#5}{#6}{code}%
+      }%
+  }
+\def\pgfparserlet@let#1#2#3#4#5#6#7%
+  {%
+    \expandafter\let\csname\pgfparser@name{#1} #2 #3 #7\expandafter\endcsname
+                    \csname\pgfparser@name{#4} #5 #6 #7\endcsname
+  }
+
+\pgfutil@protected\long\def\pgfparserdefunknown#1#2#3%
+  {%
+    \pgfparser@initiate@parser{#1}%
+    \pgfutil@namedef{\pgfparser@name{#1} #2 unknown}{#3}%
+  }
+\pgfutil@protected\long\def\pgfparserdeffinal#1#2%
+  {%
+    \pgfutil@namedef{\pgfparser@name{#1} final action}{#2}%
+  }
+\pgfutil@protected\def\pgfparserswitch#1%
+  {%
+    \pgfparser@status@state{#1}%
+    \def\pgfparserstate{#1}%
+  }
+\pgfutil@protected\def\pgfparserparse#1%
+  {%
+    \pgfutil@IfUndefined{if\pgfparser@ifname{#1 silent}}
+      {\pgferror{No parser named '#1' defined}}
+      {%
+        \edef\pgfparser@current{\pgfparser@name{#1}}%
+        \def\pgfparser@usersname{#1}%
+        \pgfparser@status@start{#1}%
+        \pgfparserswitch{initial}%
+        \pgfparser@getnexttoken
+      }%
+  }
+% get next token
+\pgfutil@protected\def\pgfparser@getnexttoken
+  {%
+    \ifx\pgfparserstate\pgfparser@final
+      \pgfparser@fi@BTb
+    \fi
+    \pgfutil@secondoftwo
+    {%
+      \pgfutil@IfUndefined{\pgfparser@current\space final action}
+        {}
+        {\csname \pgfparser@current\space final action\endcsname}%
+    }
+    {%
+      \futurelet\pgfparsertoken\pgfparser@getnexttoken@a
+    }%
+  }
+\pgfutil@protected\def\pgfparser@getnexttoken@a
+  {%
+    \pgfparser@getnexttoken@iftokengobble\bgroup\bgroup
+      {%
+        \pgfparser@getnexttoken@iftokengobble\egroup\egroup
+          {%
+            \pgfparser@getnexttoken@iftokengobble\pgfutil@sptoken{ }%
+              {\pgfparser@getnexttoken@defchar}%
+          }%
+      }%
+  }
+\pgfutil@protected\def\pgfparser@getnexttoken@iftokengobble#1#2%
+  {%
+    \ifx\pgfparsertoken#1%
+      \def\pgfparserletter{#2}%
```

Comment:
That is done because `\pgfparsertoken` is already `\let` to what ever follows, `\pgfparserletter` isn't needed by the module itself, but it might be useful for someone who writes a parser in the `\pgfparserdefunknown` code (because getting back a token from its `\meaning` is very hard so if one needs the actual letter/token this is convenient). Also `#2` might be a single space (there are 3 possibilities: `\bgroup`, `\egroup` and ` ` (space))

Reply:
It is explained in the documentation.